app/controllers/admin/admin_users_controller.rb

```diff
@@ -47,11 +47,13 @@ def set_roles
   def show
     @hide_dashboard = true
     @user = User.find_by(login: params[:id])
```

Comment:
Instead of checking `unless @user`, you can use `find_by!`, which will raise a RecordNotFound error that [Rails automatically handles with a 404](https://guides.rubyonrails.org/configuring.html#config-action-dispatch-rescue-responses). You'll need to update the test to match.

We do this in a few other places (e.g. #3412).